contrib/quicktag.lua

```diff
@@ -0,0 +1,221 @@
+--[[
+    This file is part of darktable,
+    Copyright (C) 2017 by Christian Kanzian
+
+    Thanks to
+    Copyright (C) 2016 Bill Ferguson
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+--[[
+QUICKTAG
+   For faster attaching your favorite tags, the script adds shortcuts and
+   the module "quicktag" in lighttable mode with a changeable number of buttons.
+   To each button a tag can be assigned. If the tags do not exist in your database,
+   they are added to the database once they get the first time attached to an image.
+
+   The number of buttons/shortcuts can be changed in the lua preferences.
+   Changes in the number require a restart of darktable.
+
+INSTALATION
+   * copy this file in $CONFIGDIR/lua/ where CONFIGDIR is your darktable configuration directory
+   * add the following line in the file $CONFIGDIR/luarc require "quicktag"
+
+USAGE
+   * set the number of quicktags between 2 and 10 in the preferences dialog and restart darktable
+   * if wanted set the shortcuts in the preferences dialog
+   * to add or change a quicktag, first select the old tag with the combobox "old quicktag",
+     enter a new tag in the "new quicktag" filed and press "set quicktag"
+   * use a shortcut or button to attach the tag to selected images
+
+TODO
+    * abbrevate button labels
```

Comment:
This point was raised by spaceChRiS. For me the side panel is wide enough all tags I have. Tags can be really, really long. I've seen that there are string functions in lib/dtutils. 

I want to wait for complains and add this maybe later.